pkg/storage/image.go

```diff
@@ -102,18 +104,16 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 			return nil, err
 		}
 		for _, image := range images {
-			var size *uint64
-			if len(image.Names) != 0 {
-				ref, err := svc.getRef(image.Names[0])
-				if err != nil {
-					return nil, err
-				}
-				img, err := ref.NewImage(systemContext)
-				if err != nil {
-					return nil, err
-				}
-				size = imageSize(img)
+			ref, err := istorage.Transport.ParseStoreReference(svc.store, "@"+image.ID)
+			if err != nil {
+				return err
+			}
+			img, err := ref.NewImage(systemContext)
+			if err != nil {
+				return nil, err
 			}
+			size := imageSize(img)
+			img.Close()
```

Comment:
Looking at all callers of `imageSize`, this could be simplified by having an `imageSize(types.ImageReference)` which does the `NewImage+current_imageSize+Close`. Not a blocker for this PR.